ocaml/xenopsd/xc/device.ml

```diff
@@ -3024,6 +3016,33 @@ module Backend = struct
           match params with
           | "" ->
               qmp_send_cmd domid Qmp.(Eject (cd, Some true)) |> ignore
+          | params when is_nbd params ->
+              let path, exportname = nbd params in
+              info "%s: domain=%d NBD socket=%s" __FUNCTION__ domid path ;
+              with_socket path @@ fun fd ->
+              let cmd = Qmp.(Add_fd None) in
```

Comment:
This code could be shared with the non-nbd case by defining a function and giving it the path and whether to close the FD or not as parameters.

Unless we want to completely drop the non-nbd path eventually?

Reply:
I expect that we will drop the non-nbd code eventually. If you look closely there are some difference between the two cases that are not so easy to factor out and so I am not convinced that a more complicated logic buys much.